tests/pri_queue_tests.cpp

```diff
@@ -0,0 +1,165 @@
+#include <thread>
+#include <boost/test/unit_test.hpp>
+#include <appbase/application.hpp>
+
+using namespace appbase;
+
+BOOST_AUTO_TEST_SUITE(pri_queues)
+
+constexpr static int q0 = appbase::default_queue;
+
+// start app thread and call execution loop based on use_default_queue flag 
+std::thread start_app_thread(appbase::scoped_app& app, bool use_default_queue) {
+   const char* argv[] = { boost::unit_test::framework::current_test_case().p_name->c_str() };
+   BOOST_CHECK(app->initialize(sizeof(argv) / sizeof(char*), const_cast<char**>(argv)));
+   app->startup();
+   std::thread app_thread( [&, use_default_queue]() {
+      if ( use_default_queue ) {
+         app->exec();
+      } else {
+         app->multi_queue_exec();
+      }
+   } );
+   return app_thread;
+}
+
+// wait for time out or the number of results has reached
+void wait_for_results(const std::vector<int>& results, size_t expected) {
+   auto i = 0;
+   while ( i < 10 && results.size() < expected) {
```

Comment:
`std::vector` is not thread-safe. It is not valid to check `size()` from a different thread like this. I would recommend using a `std::atomic<int>` instead of a `std::vector` and verifying total at end of test.